docs/rules/require-expect.md

```diff
@@ -0,0 +1,84 @@
+# Ensure that `expect` is called when using assertions outside of top-level test callback
+
+QUnit's `assert.expect(...)` helps developers create tests that correctly fail
+when their expected number of assertions are not called. QUnit will throw an
+error if no assertions are called by the time the test ends. This rule goes
+further and ensures that `expect` is used anytime a developer uses are assertion
+that is nested in a block or anytime a developer passes `assert` to another
+function.
+
+# Rule Details
+
+The following patterns are considered warnings:
+
+```js
+test('name', function(assert) {
+    assert.ok(true);
+    while (false) {
```

Comment:
For clarity, I'm wondering if this should be converted to an `if` condition, since that might be closer to a real use case?